tests/jsTests/indexSpec.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * Suite : Tests for showing error on incorrect field entry
+ * Description : This is a test suite for showing error in case of an invalid field entry
+ */
+describe("Tests for showing error on incorrect field entry", function() {
+	var msg = 'This is a test message for showing error';
+	beforeEach(function() {
+		setFixtures('<div class="field"> \
+				<div class="ui red pointing"></div> \
+			</div>');
+		showError($('.field'), msg);
+	});
+
+	it("should be visible", function() {
+		expect($('.field')).toHaveClass('error');
+		expect($('.ui.red.pointing')).toContainText(msg);
+		expect($('.ui.red.pointing')).toBeVisible();
+	});
+});
+
+/**
+ * Suite : Tests for hiding error on correct field entry
+ * Description : This is a test suite for hiding error in case of an valid field entry
+ */
+describe("Tests for hiding error on correct field entry", function() {
+	var msg = 'This is a test message for not showing error';
+	beforeEach(function() {
+		setFixtures('<div class="field"> \
+				<div class="ui red pointing">' + msg + '</div> \
+			</div>');
+		hideError($('.field'));
+	});
+
+	it("should be visible", function() {
+		expect($('.field')).not.toHaveClass('error');
+		expect($('.ui.red.pointing')).toBeHidden();
+	});
+});
+
+/**
+ * Suite : Tests for not empty field
+ * Description : This is a test suite for checking when field is not empty
+ */
+describe("Tests for not empty field", function() {
+	var msg = 'This is a test message when field is not empty',
+		r = false;
+	beforeEach(function() {
+		setFixtures('<div class="field">\
+			<label>Name</label>\
+			<input type="text" name="name" id="name" placeholder="Your complete name" value="Jon Snow">\
+			<div class="ui basic red left pointing prompt label transition hide"></div>\
+		</div>');
+		r = notEmpty({ ele: $('#name'), text: msg});
+	});
+
+	it("should be visible", function() {
+		expect($('.field')).not.toHaveClass('error');
+		expect($('.ui.red.pointing')).toBeHidden();
+		expect(r).toBeTruthy();
+	});
+});
+
+/**
+ * Suite : Tests for empty field
+ * Description : This is a test suite for checking when feild is empty
+ */
+describe("Tests for empty field", function() {
+	var msg = 'This is a test message when feild is empty',
+		r = false;
+	beforeEach(function() {
+		setFixtures('<div class="field">\
+			<label>Name</label>\
```

Comment:
Since the basic format (except the value="" ) is same in Test for empty and non empty field, can we make a function which would call this same fixture ? We should avoid repeating code blocks. See if there is a way you can pass a variable and use it for "value" of "name".